File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> numCheck = new HashSet<>();
        
        for(int num : nums){
            numCheck.add(num);
        }
        int maxStreak = 0;
        for(int num : numCheck){
            if(!(numCheck.contains(num-1))){
                int currNum = num;
                int streak = 1;

                while(numCheck.contains(currNum+1)){
                    currNum += 1;
                    streak += 1;
                }
                maxStreak = Math.max(maxStreak, streak);
            }
        }
        return maxStreak;
    }

}

//unsorted int arr
//len of longest consecutive elements 

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        max_count = 0

        for num in seen:
            if num-1 not in seen: #if prev num not in set, 
                current = num
                count = 1 #only 1 elem is in sequence rn

                while current+1 in seen: #count if next num in set
                    current += 1
                    count += 1
                max_count = max(max_count, count)
        return max_count




